drivers/input/input_ite_it8xxx2_kbd.c

```diff
@@ -162,6 +162,12 @@ static int it8xxx2_kbd_init(const struct device *dev)
 	/* Disable wakeup and interrupt of KSI pins before configuring */
 	it8xxx2_kbd_set_detect_mode(dev, false);
 
+	/* only supports up to 18 column lines currently */
+	if (common->col_size > 18) {
+		LOG_ERR("unsupported column size (%d > 18)", common->col_size);
+		return -ENOTSUP;
+	}
```

Comment:
there's a

`BUILD_ASSERT(IN_RANGE(DT_INST_PROP(0, col_size), 1, 18), "invalid col-size");`

at the bottom of the file, this should not be possible, drop?